File: Python/Program/Prodi.py
#   Saya Cantika Putri Arbiliansyah dengan NIM 2103727 mengerjakan soal Latihan 4 
#   dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
#   maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin

#DEKLARASI KELAS PRODI
class Prodi:
    __namaProdi = ""
    __kode = ""
    #COMPOSITE DARI KELAS COURSE
    __course = [] 

    #KONSTRUKTOR
    def __init__(self, namaProdi="",  kode="", course=[]):
        self.__namaProdi = namaProdi
        self.__kode = kode
        self.__course = []

    #GETTER DAN SETTER

    def get_namaProdi(self):
        return self.__namaProdi

    def set_namaProdi(self, namaProdi):
        self.__namaProdi = namaProdi

    def get_kode(self):
        return self.__kode

    def set_kode(self, kode):
        self.__kode = kode

    def get_course(self):
        return self.__course

    def set_course(self, course):
        self.__course.append(course)

File: CPP/Program/Prodi.cpp
//LIBRARY
#include "Header.hh"

//KONSTRUKTOR 
Prodi::Prodi()
{
    this->nama_prodi = "";
    this->kode = "";
}

Prodi::Prodi(string nama_prodi, string kode, Course course)
{
    this->nama_prodi = nama_prodi;
    this->kode = kode;
    this->course.push_back(course);
}

//SETTER GETTER NAMA PRODI
string Prodi::get_namaProdi()
{
    return this->nama_prodi;
}

void Prodi::set_namaProdi(string nama_prodi)
{
    this->nama_prodi = nama_prodi;
}

//SETTER GETTER KODE MATA KULIAH
string Prodi::get_kode()
{
    return this->kode;
}

void Prodi::set_kode(string kode)
{
    this->kode = kode;
}

//SETTER GETTER COURSE
list<Course> Prodi::get_course()
{
    return this->course;
}

void Prodi::set_course(Course course)
{
    this->course.push_back(course);
}

//DESTRUKTOR
Prodi::~Prodi()
{

}